File: py/Reverse Integer.py
class Solution:
    # @return an integer
    def reverse(self, x):
        sign = 0
        if x < 0:
            sign = 1
            x = -x
        
        l = list(str(x))
        l.reverse()
        ret = int(''.join(l))
        
        if sign == 1:
            return -ret
        return ret

File: java/Reverse Integer.java
public class Solution {
    public int reverse(int x) {
        if (x == 0)
            return x;
            
        int sign = 1;
        int iret = 0;
        
        if(x < 0){
            sign = -1;
            x = -x;
        }
        
        while(x > 0 ){
            iret = iret * 10 + (x % 10);
            x = x / 10;
        }
        
        return iret * sign;
    }
}